warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IngestTypeVisitor.java

```diff
@@ -354,25 +353,28 @@ public Set<String> getIngestTypesForField(String field) {
     }
 
     @SuppressWarnings("unchecked")
-    private Set<String> getIngestTypesForIntersection(ASTAndNode node) {
+    public Set<String> getIngestTypesForIntersection(ASTAndNode node) {
         Set<String> ingestTypes = new HashSet<>();
         for (int i = 0; i < node.jjtGetNumChildren(); i++) {
             JexlNode child = JexlASTHelper.dereference(node.jjtGetChild(i));
+
+            boolean negated = child instanceof ASTNotNode;
+            boolean isNullLiteral = child instanceof ASTEQNode && JexlASTHelper.getLiteralValueSafely(child) == null;
+            if (negated || isNullLiteral) {
+                continue;
+            }
+
             Set<String> childIngestTypes = (Set<String>) child.jjtAccept(this, null);
 
             if (childIngestTypes == null) {
-                continue; // we could have a malformed query or a query with a _Drop_ marker
+                // we could have a malformed query or a query with a _Drop_ marker
+                continue;
             }
 
             if (ingestTypes.isEmpty()) {
                 ingestTypes = childIngestTypes;
             } else {
-                if (child instanceof ASTNotNode) {
-                    // special handling of negations. negated ingest types get OR'd together
-                    ingestTypes.addAll(childIngestTypes);
-                } else {
-                    ingestTypes = intersectTypes(ingestTypes, childIngestTypes);
-                }
+                ingestTypes = intersectTypes(ingestTypes, childIngestTypes);
```

Comment:
Could you add a comment or explain why the special handling of negations for instances of ASTNNotNode's are no longer needed?

Reply:
Does the method level comment answer the question about negations, or would you like to see something more explicit?

Reply:
LGTM